test/unit/strict-template-policy.html

```diff
@@ -0,0 +1,240 @@
+          '    <div id="injected"></div>'+
+          '  </template>`' + 
+          '</dom-bind>';
+        }, /dom-bind not allowed/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-if', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-if if>' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>' + 
+            '</dom-if>';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-repeat', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-repeat items="[0]">' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>`' + 
+            '</dom-repeat>';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-module after registration', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-module id="trusted-element">' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>`' +
+            '</dom-module>' +
+            '<trusted-element></trusted-element>';
+        }, /trusted-element registered twice/);
+        const el = document.querySelector('trusted-element');
+        assert.notOk(el && el.shadowRoot && el.shadowRoot.querySelector('#injected'));
+      });
+      
+      test('dom-module before registration', function() {
+        document.getElementById('target').innerHTML =
+          '<dom-module id="has-no-template">' +
+          '  <template>' +
+          '    <div id="injected"></div>'+
+          '  </template>`' +
+          '</dom-module>';
+        class HasNoTemplate extends Polymer.Element {
+          static get is() { return 'has-no-template'; }
+          static get template() { return null; }
+        }
+        customElements.define(HasNoTemplate.is, HasNoTemplate);
+        let el = document.createElement('has-no-template');
+        document.getElementById('target').appendChild(el);
+        assert.notOk(el.shadowRoot);
+      });
+
+      test('dom-module after registration (legacy)', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-module id="trusted-element-legacy">' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>`' +
+            '</dom-module>' +
+            '<trusted-element-legacy></trusted-element-legacy>';
+        }, /trusted-element-legacy registered twice/);
+        const el = document.querySelector('trusted-element-legacy');
+        assert.notOk(el && el.shadowRoot && el.shadowRoot.querySelector('#injected'));
```

Comment:
Again, only test that `el.shadowRoot` does not exist.

Reply:
Done.